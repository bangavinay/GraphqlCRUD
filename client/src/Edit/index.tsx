import React from 'react';
import { useHistory } from "react-router-dom";
import { UPDATE_POST_MUTATION, GET_POST_QUERY } from '../Queries';

const Edit = (props) => {
    const history = useHistory();
    const [updatePost] = UPDATE_POST_MUTATION();
    const id = props.match.params.id;
    
    let titleInput, descriptionInput, activeVal;  
    let submitform = (e)=>{
      e.preventDefault();
      updatePost({ variables: { id,title: titleInput.value, description: descriptionInput.value, active: activeVal.checked } }).then(({data}:any)=>{
        console.log(data,"fdfd");
        if(data.updatePost.id){
          history.push('/list');
        }
      });
    }
    const {loading, error, data} = GET_POST_QUERY(id);
     if (loading) return <p>Loading...</p>;
     if (error) return <p>Error :(</p>;
    return(<div className ='form'>
      <form onSubmit={submitform}>
        <p>
          <label> Title:
            <input type="text" defaultValue = {data.post.title} ref={node => {
              titleInput = node;
            }} />
          </label>
        </p>
        <p>
        <label> description:
          <input type="text" defaultValue = {data.post.description} ref={node => {
              descriptionInput = node;
            }} />
        </label>
        </p>
        <p>
        <label> Is active:
          <input type="checkbox" defaultChecked={data.post.active} ref={node => {
              activeVal = node;
            }} />
        </label>
        </p>
        <button type="submit">Update Post</button>
      </form>
    </div>);
  }

  export default Edit;