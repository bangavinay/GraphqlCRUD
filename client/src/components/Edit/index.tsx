import React from 'react';
import { useHistory } from "react-router-dom";
import { UPDATE_POST_MUTATION, GET_POST_QUERY } from '../../queries';

const Edit = (props) => {
    const history = useHistory();
    const [updatePost] = UPDATE_POST_MUTATION();
    const id: number = props.match.params.id;
    
    let titleInput: HTMLInputElement, 
    descriptionInput: HTMLInputElement, 
    activeVal: HTMLInputElement;  
    /**
     * Function used to call put mutation for edit of posts
     * @param {e} Event
     */
    let editPostSubmit = (event)=>{
      event.preventDefault();
      updatePost({ variables: { id,title: titleInput.value, description: descriptionInput.value, active: activeVal.checked } }).then(({data}:any)=>{
        if(data.updatePost.id){
          history.push('/list');
        }
      });
    }
    //Fetch of post based on the id in url
    let {loading, error, data} = GET_POST_QUERY(id);
    if (loading) return <p>Loading...</p>;
    if (error) return <p>Error :(</p>;
   

    return(<div className ='form'>
      <form onSubmit={editPostSubmit}>
        <p>
          <label> Title:
            <input type="text" defaultValue = {data.post.title} ref={node => {
              titleInput = node;
            }} />
          </label>
        </p>
        <p>
        <label> description:
          <input type="text" defaultValue = {data.post.description} ref={node => {
              descriptionInput = node;
            }} />
        </label>
        </p>
        <p>
        <label> Is active:
          <input type="checkbox" defaultChecked={data.post.active} ref={node => {
              activeVal = node;
            }} />
        </label>
        </p>
        <button type="submit">Update Post</button>
      </form>
    </div>);
  }

  export default Edit;