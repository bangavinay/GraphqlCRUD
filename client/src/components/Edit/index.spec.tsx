import React from 'react';
import { render, wait, getByText } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import Edit from '.';
import { GET_POST } from '../../queries';

describe('Render Edit', () => {
    const postMock = {
        request: {
          query: GET_POST,
          variables: { id : 1 },
        },
        result: {
            data: {
              post: { id: 1, title: 'facebook', description: 'social media', active: true },
            },
          },
      };

      const props = {
          match:{
              params:{
                  id:1
              }
          }
      }
    test('Loaded State of Edit Form', async () => {
        const  { getByText } = render(
            <MockedProvider mocks={[postMock]} addTypename={false}>
                <Edit {...props} />
            </MockedProvider>);
        await wait();

        const test =await getByText(/Update /i)
        expect(test).toBeInTheDocument();
    });
});
