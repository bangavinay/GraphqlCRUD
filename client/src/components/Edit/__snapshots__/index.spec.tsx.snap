// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Edit Loading State of Edit Form 1`] = `
<p>
  Loading...
</p>
`;
